src/utilities/checkDuplicateKeys.ts

```diff
@@ -1,6 +1,6 @@
 export function checkDuplicateKeys(aParams: Record<string, unknown>, bParams: Record<string, unknown>): { key: string, hasDuplicates: true } | { key: undefined, hasDuplicates: false } {
-  const aParamKeys = Object.keys(aParams)
-  const bParamKeys = Object.keys(bParams)
+  const aParamKeys = Object.keys(aParams).map(removeLeadingQuestionMark)
+  const bParamKeys = Object.keys(bParams).map(removeLeadingQuestionMark)
```

Comment:
don't love this, but I think it's necessary because technically params could theoretically be stored with and without the `?`. However, I've made the decision to keep our rule of not duplicate param names.